docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,279 @@
+
+Thus, like the status tables above, updates to keys in this info table will instead be an insert of the new info paired with deletion of any older revision of it. This case is actually simpler than the status case as no history is required and the addition of MVCC-like behaviour is solely for making past revisions splittable, so each insert can delete all prior revisions.
+
+#### Mutual Exclusion of Updates
+
+Updates to system.job_info should, by default, only be allowed from processors that were started as part of the currently running instance of the job. The API for updating the system.job_info, system.job_status_progress, and system.job_status_message tables will require the current claim_session_id for the job.  This session ID will be passed from the job system to the job implementation’s relevant methods (e.g. Resume). Jobs that wish to update job info from multiple nodes will need to pass the session ID to the relevant processors on those nodes.
+
+Updates will be allowed so long as the session specified by the session ID is still alive:
+
+```
+INSERT INTO system.jobs_info SELECT $1, $3, NOW(), $4 WHERE crdb_internal.sql_liveness_is_alive($2) IS TRUE
+```
+
+Since jobs are only eligible for resumption once the existing session ID returns false from crdb_internal.sql_liveness_is_alive, such a condition should ensure that we are not updating the job_info  table while another instance of the job is running without going to the jobs_control table.
+
+##### Alternatives options for Update Exclusion 
+
+We could allow a limited violation of our invariant and do a join on the system.jobs_control table:
+
+```
+INSERT INTO system.jobs_info SELECT id, $3, NOW(), $4 FROM system.jobs_control WHERE id = $2 AND claim_session_id = $2;
+```
+
+While this would be touching the jobs_control table, it would be a point read on a single row and would not contend with scans of the jobs table. If testing proves contention-free at scale, we may prefer this implementation.
```

Comment:
Sounds like you're already planning to do this, but I highly recommend setting up some sort of prototype environment to validate the choices being made in this RFC and to verify they work at scale as expected. This can be done as a standalone binary using an existing CRDB cluster (i.e. not baking the code into CRDB). We did this in one perf critical piece of code in CC and the extra effort of the prototype saved a lot of time.